Feature/zcnsc

Dmdv opened the pull request.

Dmdv committed:
Added IncreaseAllowance and ERC20 contract binding

Dmdv committed:
Adding burn functions

Dmdv committed:
Ported bridge binding

Dmdv committed:
Merge branch 'staging' into feature/zcnsc

Dmdv committed:
Fixed dependencies

Dmdv committed:
Added GetAuthorizer method for SDK

Dmdv committed:
Added burn tickets collection

Dmdv committed:
Burn ticket reponse, parsing and fetching to client

Dmdv committed:
MainPayload tests

Dmdv committed:
Refactoring

Dmdv committed:
Updated config and refactoring

Dmdv committed:
ZCNBridge initialization

Dmdv committed:
Adde node and fix imports

Dmdv committed:
Node start

Dmdv committed:
Init bridge

Dmdv committed:
Completed test and refactoring before the minting

Dmdv committed:
Added minting

Dmdv committed:
Refactoring

Dmdv committed:
Added Burn and refctored Mint functions

Dmdv committed:
Renaming jobs results

Dmdv committed:
Refactoring of http client

Dmdv committed:
Renaming

Dmdv committed:
Refactoring

Dmdv committed:
Added ZCN burn ticket

Dmdv committed:
Authorizer generic job

Dmdv committed:
Refactored minting

Dmdv committed:
Renaming

Dmdv committed:
Moved payloads to zcnsc folder

Dmdv committed:
Added Mint method from Ethereum Bridge contract

moldis commented on constants/files.go:
```diff
@@ -2,7 +2,7 @@ package constants
 const (
 	// FileOperationInsert method name of add new file
-	FileOperationInsert      = "insert"
+	FileOperationInsert      =x "insert"
```
@Dmdv Seems error here

Dmdv replied:
Fixed

moldis commented on .golangci.yml:
```diff
@@ -0,0 +1,79 @@
+linters-settings:
```
@Dmdv I think this file should be in different branch

Dmdv replied:
Fixed

moldis commented on zcnbridge/authorizers_query.go:
```diff
@@ -0,0 +1,213 @@
+			return nil, errors.Wrap("type_cast", "failed to convert to *proofEthereumBurn", err)
+		}
+
+		var sigs []*zcnsc.AuthorizerSignature
```
As I understood Authorizers should authorize burn tickets, how long it will take if the client calls it? And how many Authorizers required?

Dmdv replied:
It depends on the transaction confirmed. If Ethereum transaction is confirmed then this will be fast 1-10 seconds. For ethereum transaction this is fast. We will have for start 20 authorizers. All authorizer requests will be run async concurrently @moldis

moldis replied:
@Dmdv Even 10 seconds is slow, and I guess it might take more time and not depend on us, more on eth network. Any way we can move this to the backend side and let client just sign the transaction, then give back some sort of hash to check transactions status?

moldis commented on zcnbridge/bridge.go:
```diff
@@ -0,0 +1,334 @@
+		Data: data,
+	})
+	if err != nil {
+		zcncore.Logger.Fatal(err)
```
Wrong return value

Dmdv replied:
Fixed

Dmdv committed:
Temp disabled tests

Dmdv committed:
Merge branch 'staging' into feature/zcnsc

Dmdv committed:
Moving lint rules to another branch

Dmdv committed:
Fixed typos

Dmdv committed:
Updated dependencies

Dmdv committed:
Updated dependencies

Dmdv committed:
Minor fixes and refactoring

Dmdv committed:
Minor fixes and refactoring

Dmdv committed:
return fix

Dmdv committed:
Renaming

Dmdv committed:
Refactored calculation

Dmdv committed:
Configuration update

Dmdv committed:
Fixing cmd and config parameters

Dmdv committed:
Updated response from Authorizer

Dmdv committed:
Merge branch 'staging' into feature/zcnsc

Dmdv committed:
Collecting tickets from ZCN transaction

Dmdv committed:
Updated keys config

Dmdv committed:
Removed some params from config, improved gas calculation, some refactoring

Dmdv committed:
Tested IncreaseAllowance

Dmdv committed:
Improved zcncore initialization and ethereum client

Dmdv committed:
Eth wallet instance moved to node;

Rethinking and redesigning of how gas estimations should be made;
Fixed some minor issues;
Improved performance of Eth transactions

Dmdv committed:
Minting gas price refactoring

Dmdv committed:
Added authorizer sample app tests

Dmdv committed:
Merge branch 'staging' into feature/zcnsc

Dmdv committed:
Renaming

Dmdv committed:
Updated example:

1. Setup wallet using wallet generated with zwalletcli
2. Refactored the client and initialization

Dmdv committed:
Added confirmation tests

Dmdv committed:
Cleaning client config

Dmdv committed:
Cleaned configuration

Dmdv committed:
Refactored the API and example

Dmdv committed:
Refactored status/progress for registration

Dmdv committed:
Refactored API configuration

Dmdv committed:
Refactored, removed node, improved the API, and solved minor issues
